### PR TITLE
feat: Add scroll snapping to responsive tabs

### PR DESCRIPTION
**Pull request by timogasda**
### Description

This PR includes two changes to the tabs component:

1. Adds CSS scroll snapping to the responsive tabs, which improves the experience on touch devices and reduces the risks of tab titles being "skipped" when using the left/right navigation buttons.
2. Removed the custom implementation of smooth scrolling, since all our supported browsers can do this natively now.

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/tabs/smooth-scroll.ts [hunk 1]
```diff
@@ -2,62 +2,16 @@
...
-
 const smoothScroll = (element: HTMLElement, to: number) => {
-  if (isMotionDisabled(element)) {
+  if (isMotionDisabled(element) || !element.scrollTo) {
     element.scrollLeft = to;
-    return;
-  }
-  if (isNativeSmoothScrollingSupported() && element.scrollTo) {
+  } else {
+    // istanbul ignore next: unit tests always have motion disabled
```
Comment (timogasda) [inline src/tabs/smooth-scroll.ts:9]: I don't understand why this is not working properly. If I move the `ignore` comment above line 8, it still complains. I think we might just need to override the coverage check for this PR.

Comment (timogasda) [reply src/tabs/smooth-scroll.ts:9]: (if this becomes a blocker, I'm happy to separate this out into a second PR. I just really need to deliver the scrolling update)

Comment (connorlanigan) [reply src/tabs/smooth-scroll.ts:9]: Have you tried using `/* istanbul ignore else  */`? https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md#ignore-an-else-path

**Review by connorlanigan — APPROVED** [review]

File: src/tabs/smooth-scroll.ts (hunk 1 above)
Comment (timogasda) [reply src/tabs/smooth-scroll.ts:9]: I didn't, but according to the docs that's for cases where an `else` path might be required/expected but it doesn't exist at all.

**Pull request merged**